app/Models/Message.php

```diff
@@ -37,4 +37,16 @@ public static function getTypes()
     {
         return static::$types;
     }
+
+    public static function prepareMessage($message, $model)
+    {
+        //@TODO: I only tested this with competition, is there other things we want to replace?
+        $tokens = $model::$tokenizable;
+        foreach ($tokens as $token) {
+            $message->body = str_replace(':'.$token, $model->$token, $message->body);
```

Comment:
Could we abstract this into a function that just takes an array of tokens and a string of text, so as we need to add more models and tokens we can just combine them all into one array and then use this new function to do the string replacement?